File: 1-ADT/contador/contador.js
/*
  Implementacion del ADT Contador

  @author: Jorge Londono

  Para ejecutar: node contador.js

 */

/*
    Esta forma de importar modulos require ES6
    Se debe definir un package.json que contenga la especificacion: "type": "module"
*/
import assert from 'assert';


function Contador(id) {
    this.conteo = 0;
    this.id = id;
    this.incrementar = function() { this.conteo++; };
    this.getConteo = function() { return this.conteo; };
    this.getId = function() { return this.id; };
}


// Ejemplos de uso
let c = new Contador('Eventos');
assert.strictEqual(c.conteo, 0, 'El conteo inicial debe ser cero');
c.incrementar();
c.incrementar();
assert.strictEqual(c.conteo, 2, 'El conteo inicial debe ser 2');
assert.strictEqual(c.id, 'Eventos');
console.log(`El contador ${c.id} tiene un valor de ${c.conteo}`);

File: 1-ADT/contador/Contador.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-


class Contador:
    '''Implementacion del ADT Contador como una clase en Python
    '''

    def __init__(self, id):
        self._id = id
        self._conteo = 0

    def incrementar(self):
        self._conteo += 1

    def getConteo(self):
        return self._conteo

    def getId(self):
        return self._id


# Prueba unitaria del ADT Contador
if __name__ == '__main__':
    a = Contador('prueba')
    assert a.getConteo() == 0, "El valor inicial debe ser cero"
    a.incrementar()
    a.incrementar()
    assert a.getConteo() == 2, "El valor del conteo debe ser dos"
    assert a.getId() == 'prueba'
